### PR TITLE
SDIT-937 Add prisoner-search files

### PR DESCRIPTION
**Pull request by karenmillermoj**
Prisoner search files pulled in from prisoner-offender-search (and modified for opensearch)

### CODE REVIEW

File: hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/services/dto/PhysicalDetailRequest.kt [hunk 1]
```diff
@@ -0,0 +1,162 @@
+package uk.gov.justice.digital.hmpps.prisonersearch.search.services.dto
+
+import io.swagger.v3.oas.annotations.media.Schema
+
```
Comment (karenmillermoj) [inline hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/services/dto/PhysicalDetailRequest.kt:4]: This is file is part of the next PR (accidentally included it - but will not affect anything)

File: hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/config/GsonConfig.kt [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import java.lang.reflect.Type
+import java.time.LocalDate
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
+import java.util.Locale
+
+@Configuration
+class GsonConfig() {
```
Comment (petergphillips) [inline hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/config/GsonConfig.kt:20]: I actually removed this file from indexer and switched to object mapper instead.  Jackson is the default with spring boot so made more sense to just use that out of the box

File: hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/services/dto/PhysicalDetailRequest.kt [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+data class PhysicalDetailRequest(
+  @Schema(
+    description = "List of prison codes to filter results by",
+    example = "['LEI', 'MDI']",
+    required = true,
```
Comment (petergphillips) [inline hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/services/dto/PhysicalDetailRequest.kt:9]: think this `required` is now deprecated?  Replaced with `RequiredMode`?

**Review by petergphillips — APPROVED** [review]
LGTM

**Pull request merged**